### ISSUE TITLE
Compiling 1.6.8 fails on Ubuntu 18.04 and 22.04

### ISSUE THREAD

**Issue by gitterdude** [drobilla/jalv#55]
I get this error 

[62/64] Compiling C++ object jalv.qt5.p/meson-generated_.._jalv_qt5_meta.cpp.o
FAILED: jalv.qt5.p/meson-generated_.._jalv_qt5_meta.cpp.o 
/usr/include/x86_64-linux-gnu/qt5/QtCore/qglobal.h:1281:4: error: #error "You must build your code with position independent code if Qt was built with -reduce-relocations. " "Compile your code with -fPIC (and not with -fPIE)."

Below is my quick-n-dirty hack to fix it. There's probably a more correct way of  doing it, though.

```
--- meson.build~	2022-09-20 11:23:39.000000000 +0200
+++ meson.build	2022-09-20 11:31:55.834705268 +0200
@@ -331,6 +331,7 @@
   endif
 
   if moc.found()
+    platform_defines += ['-fPIC']
     jalv_qt_hpp = files(jalv_src_root / 'src' / 'jalv_qt.hpp')
 
     jalv_qt5_meta_cpp = custom_target(
```

**Comment by drobilla** [comment]
Trouble is, this is distro-specific.  I'm not sure if there are any consequences from always doing this.

Note that you can do it from the command-line by setting cpp_args.  Although I of course agree it should build out of the box; I'll look into it.

**Comment by anagram3k** [comment]
You can execute this line from the configured build directory:
```
meson configure -Dcpp_args="-fPIC"
```
and the build will succeed. 

Maybe a simple comment in the README indicating this "fix" should suffice.

**Comment by drobilla** [comment]
> Maybe a simple comment in the README indicating this "fix" should suffice.

Oh, sweet summer child ;)

Presumably fixed in d7266c6

**Comment by anagram3k** [comment]
I found how you can detect this thing in meson:

```  
qt5_core_dep = dependency(
    'Qt5Core',
    required: get_option('qt5'),
)
qt5_reduce_relocations = false
qt5_core_config = qt5_core_dep.get_variable('qt_config')
if qt5_core_config.contains('reduce_relocations')
    qt5_reduce_relocations = true
endif
```
I already tested it on my Ubuntu 22.10 system and it works. Feel free to ignore it if it is too much of a hassle. 

I can generate a pull request if you want, but I am still learning meson. You probably can do a much better code.

sources:
[1] I got the idea from https://dev.gnupg.org/T4982
[2] Found this on the meson docs https://mesonbuild.com/Reference-manual_returned_dep.html#depget_variable
